hydra/app/extensions/clock.js

```diff
@@ -0,0 +1,47 @@
+class Clock {
+  constructor({
+    startAt = 0, 
+    countBy = 1,
+    countWith = null,
+    interval = 1000, 
+    onTick = () => {}
+  } = {}) {
+    this.currentVal = startAt
+    this.countBy = countBy
+    this.countWith = countWith 
+    ? countWith 
+    : last => last + countBy
+    this.interval = interval
+    this.onTick = onTick
+    this.counter = setInterval(
+      () => {
+        this.currentVal = this.countWith(this.currentVal)
+        this.onTick(this.currentVal)
+      }, 
+      interval) // maybe add option for setting interval with bpm, i.e. 1 is one beat, 0.5 half a beat, etc.
```

Comment:
what do you think of this?

Reply:
hmmm yea i am not sure, most of hydra uses requestAnimationFrame, and receives dt (change in miliseconds since the last update) as a parameter so that everything is a function of time. nott sure how that will work with setInterval!